agent/worker.go

```diff
@@ -219,6 +256,39 @@ func reconcileTaskState(ctx context.Context, w *worker, added []*api.Task, remov
 	return tx.Commit()
 }
 
+func reconcileSecrets(ctx context.Context, w *worker, assignments []*api.AssignmentChange, fullSnapshot bool) error {
+	var (
+		updatedSecrets []api.Secret
```

Comment:
Why does this use `[]api.Secrets` as opposed to `reconcileTaskState` which uses `[]*api.Task`? Not saying it's a problem, but wondering if there's a reason for storing pointers in one case and copies of the object in another.